### PR TITLE
Autogenerating Documentation through JSDoc

### PR DESCRIPTION
**Pull request by hanki-kim**
Created the pipeline to autogenerate documentation with JSDocs. Not the prettiest but it has the bare minimum. The files for the documentation gets pushed to the `gh-pages` branch and so our repo's GitHub Pages will be referring to that branch.
In order for JSDoc to generate documentation for the functions, we need to go through and write documentation for the individual functions inside the js files.

EX: 
```
/**
 * Represents a book.
 * @constructor
 * @param {string} title - The title of the book.
 * @param {string} author - The author of the book.
 */
function Book(title, author) {
}
```
OR
```
/**
 * Gets the sum of two numbers
 * @param {Number} a - the first operand
 * @param {Number} b - the second operand
 * @returns the sum of a + b
 */
function sum(a, b) {
    return a + b;
}
```

### CODE REVIEW

**Comment by thucngyyen** [comment]
Looks very promising!

File: package.json [hunk 1]
```diff
@@ -6,11 +6,15 @@
...
-        "standard": "*",
         "json-ptr": ">=3.0.0",
-        "json-schema": ">=0.4.0"
+        "json-schema": ">=0.4.0",
+        "standard": "*"
     },
     "scripts": {
-        "styling": "standard --fix"
+        "styling": "standard --fix",
+        "doc": "jsdoc -c conf.json"
```
[the remark this answers is not in the document]
Comment (hanki-kim) [reply package.json:15]: We do not have to. This was just to see if we can create files into the same branch you're working on instead onto gh-pages but I figured you didn't need to just to keep it simple and safe

[the remark this answers is not in the document]
Comment (hanki-kim) [reply package.json]: > 

We do not have to. This was just to see if we can create files into the same branch you're working on instead onto gh-pages but I figured you didn't need to just to keep it simple and safe. I'll remove it actually

[the remark this answers is not in the document]
Comment (thucngyyen) [reply package.json]: ok, I think we're good to go. Make sure you squash and merge

**Pull request merged**